src/Stratis.SmartContracts.Executor.Reflection/SmartContractExecutionContext.cs

```diff
@@ -53,6 +61,14 @@ public interface ITransactionContext
         /// The destination address
         /// </summary>
         uint160 To { get; }
+
+        /// <summary>
+        /// Get the next number to use as the nonce in the contract address calculation and then 
+        /// increment the number behind the scenes, so that the next contract address calculation
+        /// gets a new number. 
+        /// </summary>
+        /// <returns>The next nonce to use in the contract address formula.</returns>
+        ulong GetNonceAndIncrement();
```

Comment:
Does `IncrementNonce()` sound better ?

Reply:
Agree with @ferdeen. The method really returns the incremented nonce.

Reply:
It returns the nonce, then increments.

Reply:
So it does, all good

Reply:
If the calling code really needs to know that it will be incremented after then the method seems fine.  If it doesn't care then it could be called GetNonce(),  or even a Nonce Get property.